get_github_coderabbit_review/coderabbit_fetcher/validation.py

```
@@ -0,0 +1,659 @@
+
+            def timeout_handler_func(signum, frame):
+                raise TimeoutError(f"Function {func.__name__} timed out after {timeout_seconds}s")
+
+            # Set up the timeout
+            old_handler = signal.signal(signal.SIGALRM, timeout_handler_func)
+            signal.setitimer(signal.ITIMER_REAL, float(timeout_seconds))
+
+            try:
+                result = func(*args, **kwargs)
+                signal.setitimer(signal.ITIMER_REAL, 0)  # Cancel the timer
+                return result
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)